app/shared_channel_notifier.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+// NotifySharedChannelSync signals the syncService to start syncing shared channel updates
+func (a *App) NotifySharedChannelSync(channel *model.Channel, event string) {
+	if channel.IsShared() {
+		syncService := a.srv.GetSharedChannelSyncService()
+
+		// When the sync service is running on the node, trigger syncing without broadcasting
+		if syncService != nil && syncService.Active() {
+			mlog.Debug(
+				"Notifying shared channel sync service",
+				mlog.String("channel_id", channel.Id),
+				mlog.String("event", event),
+			)
+			syncService.NotifyChannelChanged(channel.Id)
+			return
+		}
+
+		// When the sync service is not running on the node and cluster is enabled, broadcast shared sync message
+		if a.Cluster() != nil {
+			mlog.Debug(
+				"Shared channel sync service is not running on this node. Broadcasting sync cluster event.",
+				mlog.String("channel_id", channel.Id),
+				mlog.String("event", event),
+			)
+
+			a.Cluster().SendClusterMessage(&model.ClusterMessage{
+				Event:    model.CLUSTER_EVENT_SYNC_SHARED_CHANNEL,
```

Comment:
Is there an existing event that gets broadcast on each post?  I'm concerned we just doubled cluster traffic, which was already a pain point.

Reply:
That is such a good point. Silly me not relying on the initial events, since we just need to react to those events, which already exist for post(create, update, delete) and reaction(add, remove). Will amend.

Reply:
Amended in https://github.com/mattermost/mattermost-server/pull/16675/commits/422512b78c0bce9a9281bb4b82ed0da134b04897